### PR TITLE
Allow custom JSON pointer class

### PR DESCRIPTION
**Pull request by tzoiker**
@stefankoegl It would be nice to be able to pass a custom implementation of the JSON pointer.

### CODE REVIEW

**Comment by tzoiker** [comment]
@stefankoegl is it possible to accept soon? I want to use it in another project and avoid even temporary copy-pasting.

**Review by Alanscut — APPROVED** [review]
LGTM, would you mind adding more tests for custom Pointer which is not extended from `python-json-pointer`?
Just for curiosity, can't `python-json-pointer` meet your current requirment?

File: tests.py [hunk 1]
```diff
@@ -671,6 +671,94 @@ def test_create_with_pointer(self):
...
+        self.assertTrue(patch._ops)
+        for op in patch._ops:
+            self.assertIsInstance(op.pointer, self.CustomJsonPointer)
+            self.assertEqual(op.pointer_cls, self.CustomJsonPointer)
+
+    class PrefixJsonPointer(jsonpointer.JsonPointer):
+        def __init__(self, pointer):
+            super().__init__('/foo/bar' + pointer)
+
+    def test_json_patch_wtih_prefix_pointer(self):
```
Comment (Alanscut) [inline tests.py:755]:
```suggestion
    def test_json_patch_with_prefix_pointer(self):
```
just a typo

Comment (tzoiker) [reply tests.py]: \+

**Comment by tzoiker** [comment]
> LGTM, would you mind adding more tests for custom Pointer which is not extended from `python-json-pointer`?
> Just for curiosity, can't `python-json-pointer` meet your current requirment?

Added a toy example. The idea was to at least be able to extend (or replace with the same, but custom implementation in C or whatever). In my particular case, I wanted to add an extension (out of spec, obviously) that allows to dynamically match array indexes based on the nested values of array elements.

**Comment by Alanscut** [comment]
Nice work! it will make this lib more flexiable.

File: jsonpatch.py [hunk 2]
```diff
@@ -56,7 +56,7 @@
 
 # Will be parsed by setup.py to determine package metadata
 __author__ = 'Stefan Kögl <stefan@skoegl.net>'
-__version__ = '1.24'
+__version__ = '1.27'
```
Comment (stefankoegl) [inline jsonpatch.py:59]: Please do not change the version number as part of the pull-reques. That will be done when a new release is prepared.

**Comment by stefankoegl** [comment]
@tzoiker, looks good. Please revert the change of version number - that will be done when a new release is prepared. 

Could you please also briefly describe your use case?

**Review by stefankoegl — CHANGES_REQUESTED** [review]

File: jsonpatch.py (hunk 2 above)
Comment (tzoiker) [reply jsonpatch.py]: \+

**Comment by tzoiker** [comment]
> @tzoiker, looks good. Please revert the change of version number - that will be done when a new release is prepared.
> 
> Could you please also briefly describe your use case?

Described it in this comment https://github.com/stefankoegl/python-json-patch/pull/114#issuecomment-728773802

**Review by stefankoegl — CHANGES_REQUESTED** [review]
Looks good now. I just merged another pull request, so there's a conflict now. If you can resolve that, I can merge right afterwards.

**Comment by stefankoegl** [comment]
I have just released v1.27 which inlcudes this PR. Thanks for the contribution!

**Pull request merged**